File: BOJ/1007/1007.py
import itertools
import math

t = int(input())
for i in range(0, t) :
    n = int(input())
    arr = [];
    for j in range(n) :
        x,y= input().split()
        arr.append([int(x), int(y)])
    
    min_dist = 987654321
    
    for subset in itertools.combinations(arr, int(n/2)):
        sum_x,sum_y = float(0), float(0);
        
        for arr_elem in arr :
            sum_x+= arr_elem[0];
            sum_y += arr_elem[1];
        
        for arr_elem in subset :
            sum_x -= 2*arr_elem[0];
            sum_y -= 2*arr_elem[1];
        
        
        dist = float(math.sqrt(pow(sum_x,2)+pow(sum_y,2)))    
        
        if(dist < min_dist) :
            min_dist = dist;
    
    print(min_dist);

File: BOJ/1007/1007.cpp
#include <iostream>
#include <vector>
#include <math.h>
#include <algorithm>
#include <queue>

using namespace std;

vector<pair<int,int>> p(25);
vector<bool> chk(25);
double minDist = 9876543219;

double dist(pair<int,int> a, pair<int,int> b) {
	return sqrt(pow(a.first-b.first,2)+pow(a.second-b.second,2));
}

void combinationDist(int pair, double sumDist,int n) {
	if(pair==n) {
		minDist = min(minDist , sumDist);
		return;
	}
	for(int i = 1; i <= n; i++) {
		if(chk[i]==false) {
			chk[i]=true;
			for(int j = i+1; j <= n; j++) {
				if(chk[j]==false) {
					chk[j]=true;
					cout<<"comb "<<pair<<" "<<i<<" "<<j<<endl;
					combinationDist(pair+2, sumDist+ dist(p[i],p[j]),n);
					chk[j]=false;
				}
			}
			chk[i]=false;
		}
	}
}

int main(void) {
	cin.tie(NULL);cout.tie(NULL);ios_base::sync_with_stdio(false);
	freopen("/workspace/PS_Git/BOJ/input.txt","r",stdin);
	cout<<fixed;
	cout.precision(12);
	
	int t, n;
	cin>>t;
	while(t--) {
		cin>>n;
		for(int i =1; i<= n; i++) cin>>p[i].first>>p[i].second;
		
		double answer = 0;
		minDist = 9876543219;
		
		combinationDist(0,0,n);
		answer = minDist;
		cout<<answer<<'\n';
	}
	return 0;
}